common/models/user.js

```diff
@@ -652,17 +657,18 @@ module.exports = function(User) {
           );
         }
       })
-      .flatMap(() => {
-        const update = {
-            newEmail,
-            emailVerified: false,
-            emailVerifyTTL: new Date()
-          };
-        return this.update$(update)
-          .do(() => Object.assign(this, update))
-          .flatMap(() => this.requestAuthEmail(false, newEmail));
+      .flatMap(()=>{
+        return Observable.forkJoin(
+          this.update$(updateConfig),
+          this.requestAuthEmail(false, newEmail),
+          (user, message) => ({ user, message })
```

Comment:
Seeing as we are not using the `user`, we can just skip over it, Then we are just returning the message string. so we can refactor this to not even need the `map`. We also need to update `this` at the same time, I 'think' this is some weird loopback 'in-memory' thing where we can update the db, but if we want to use the updates any time soon, we need to update `this` also.

```js
return Observable.forkJoin(
  this.update$(updateConfig),
  this.requestAuthEmail(false, newEmail),
  (_, message) => message
)
.do(() => {
  Object.assign(this, updateConfig);
});
```